warehouse/query-core/src/main/java/datawave/query/jexl/functions/IdentityAggregator.java

```diff
@@ -104,8 +107,11 @@ public Key apply(SortedKeyValueIterator<Key,Value> itr, Document doc, AttributeF
             boolean toKeep = (fieldsToKeep == null || fieldsToKeep.contains(JexlASTHelper.removeGroupingContext(fieldNameValue.first())))
                             && (filter == null || filter.keep(topKey));
             attr.setToKeep(toKeep);
-            doc.put(fieldNameValue.first(), attr);
-            key = nextKey;
+            
+            // Anything that is being kept has to be added to the doc to be returned, if we aren't keeping only add to the doc if necessary for evaluation
+            if (toKeep || (filter == null || filter.apply(new AbstractMap.SimpleEntry<>(topKey, null)))) {
+                doc.put(fieldNameValue.first(), attr);
```

Comment:
I am still confused by this.  the apply method means we need it in the document (either for evaluation or to send back to the user).  Keep is only the determination as to whether it is actually sent back to the user.  I would think filter.apply is enough here.

Reply:
The apply method only means we need it in the document, not that it needs to be sent back to the user. keep and apply are potentially mutually exclusive. This check will ensure we don't add something that is not necessary for either accidentally. One case I need to revisit is if its possible to have an index fully satisfied query, and how this would behave in that situation

Reply:
are you good with this?

Reply:
I am good with this